### ISSUE TITLE
Is there a wrapper for "NgbNav"? 

### ISSUE THREAD

**Issue by harshit-gupta2** [ng-bootstrap/ng-bootstrap#3837]
### Bug description: I was able to pass NgbTabSet from one controller to another, enclosing it inside an "ng-template" tag and assigning an id to this ng-template. But this doesn't seem to be possible with NgbNav as I can't catch NgbNav as @ViewChild, if I enclosed it inside ng-template.

This is how it was with NgbTabSet :-
```html
<ng-template #caseContent>
    <ngb-tabset #t="ngbTabset" type="pills" (tabChange)="changeTab($event)">
      <ngb-tab id="caseDetailsTab" title="Case Details" translate>
        <ng-template ngbTabContent>
          <app-case-details [unknownItemsCount]="unknownItemsCount" (viewAdditionalCase)="generateNewCaseTab($event)" [caseId]="caseId" [permissionList]="permissionList" (isQuarantined)="isQuarantined = $event" (isReviewed)="isReviewed = $event" (noteCountUpdate)="noteCount = $event"></app-case-details>
        </ng-template>
      </ngb-tab>
 </ng-template>
```

Following is corresponding component.ts :-
```ts
@ViewChild('caseContent') caseContentTemplate: any;
  @ViewChild(NgbTabset) set content(content: NgbTabset) {
    this.tabSet = content;
  }
```

Here this.tabSet would give me access to the tabs in case I want to play around with activeId or something but I can simply pass "caseContentTemplate" to another component where I actually want to display my tabs.

However, this is my NgbNav so far :-
```html
<ul ngbNav #nav="ngbNav" class="nav-tabs" (navChange)="changeTab($event)">
    <li ngbNavItem="caseDetailsTab">
      <a ngbNavLink i18n>Case Details</a>
      <ng-template ngbNavContent>
        <app-case-details [unknownItemsCount]="unknownItemsCount" (viewAdditionalCase)="generateNewCaseTab($event)"
          [caseId]="caseId" [permissionList]="permissionList" (isQuarantined)="isQuarantined = $event"
          (isReviewed)="isReviewed = $event" (noteCountUpdate)="noteCount = $event"></app-case-details>
      </ng-template>
    </li>
 </ul>

```
In component.ts :-
```ts
 @ViewChild('nav', { static: true}) set content(content: NgbNav) {
    this.tabSet = content;
  }
```

And then I am passing "this.tabSet" to some other component (as enclosing it inside <ng-template> wont work) but somehow it doesn't take NgbNavItem with it and just renders the NgbNavContent on the destination component's html page.

 Is there a workaround to be able to contain whole NgbNav and pass it to another controller? Thanks in advance.

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular: 9.1.12

ng-bootstrap: 5.6.1

@ng-bootstrap/ng-bootstrap: 6.2.0

Bootstrap: 4.5.1

**Comment by benouat** [comment]
Hey @harshit-gupta2 

I am sorry but I did not get what your issue is. I read it twice, but still, I am struggling understanding the root problem.

Would you mind replicating in a StackBlitz what you used to be doing with `NgbTabset` ? It will probably light my candle

**Comment by harshit-gupta2** [comment]
Hi @benouat,

Thank you for your response. I was just looking for a wrapper for NgbNav. I just wanted to render NgbNav on a different screen so I had placed NgbNavOutlet on a different HTML (not the same as NgbNav). We were able to achieve this with NgbTabSet, if you check my code above, NgbTabSet is inside Ng-template and this template is being rendered on a different HTML via NgTemplateOutlet. When I migrated to NgbNav, I replaced NgbTabSet with NgbNav and NgTemplateOutlet with NgbNavOutlet, which didn't work.

I finally found a fix. The trick was to have NgbNavOutlet on the same HTML where your NgbNav is. Wrap it around a ng-template and render on a different page using NgTemplateOutlet only (as it was being done with NgbTabSet). So the trick was not to get rid of NgTemplate, but just place NgbNavOutlet on the same html as your NGbNav. 

I am a little busy with some tasks at hand so I won't be able to prepare a working Slackblitz at the moment but Thank you.
